base/array.jl

```diff
@@ -1061,13 +1051,13 @@ julia> findnext(isodd, A, 2)
 0
 ```
 """
-function findnext(testf::Function, A, start::Integer)
-    for i = start:length(A)
-        if testf(A[i])
-            return i
-        end
+function findnext(testf::Function, A, idx::Integer)
+    while !done(A, idx)
```

Comment:
Would `state2ind` and `ind2state` functions fix this, like `sub2ind` and `ind2sub`?